plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/TypeCoercer.java

```diff
@@ -49,4 +49,14 @@ public Block apply(Block block)
     }
```

Comment:
You can consider implementing an optimized path for dictionary block in TypeCoercer#apply. Also for RLE if that code can take a partitioned column's block as input.

Reply:
But `dictionary block` is applicable for varchar-varchar coercion right ?

Reply:
I was thinking on having it as a follow-up as it could benefit for other coercions too. WDYT ?

In case of `DictionaryBlock` is there is any specific reason on why `ids` is not exposed in a public API ?

Reply:
Doing it as follow-up is good.
You could add a public method to BlockUtil for applying transform to dictionary block and make use of non-public APIs internally in that if it makes sense.